strapi/strapi
# Fix: media library image signedUrl cache busting issue

### What does it do?

Adds a new property called isUrlSigned to the admin upload file object, within the signFileUrls method. This property is later utilised in the ImageAssetCard component to determine whether cache-busting is required for the thumbnail URL.

### Why is it needed?

Currently, when using the AWS S3 provider with [AWS Signature Version 4 authentication with query params](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html), images in the Asset Media Library appear broken. This is due to a "SignatureDoesNotMatch" error caused by the presence of the "updatedAt" parameter, which is used for cache busting. This error occurs because the "updatedAt" parameter is not included in the canonical URL for generating the AWS S3 signature.

This pull request adds the updatedAt cache busting parameter only when the thumbnail URLs are not signed, which fixes the issue. 

### How to test it?

- Configure media library S3 provider to use v4 authentication, then upload an image in the media library. The image shouldn't appear broken.
- Add images with and without the v4 S3 provider. Check the thumbnail urls - only the ones uploaded without S3 should have the ?updatedAt query param.

### Related issue(s)/PR(s)

Related to #18443
